### PR TITLE
OneToN contract no longer works with unregistered services

### PR DESCRIPTION
**Pull request by pirapira**
This is a part of #1147.

### CODE REVIEW

File: raiden_contracts/tests/test_print_gas.py [hunk 1]
```diff
@@ -260,6 +260,15 @@ def print_gas_monitoring_service(
...
 
+    # B registers itself as a service provider
+    deposit = service_registry.functions.currentPrice().call()
+    custom_token.functions.mint(deposit).call_and_transact({"from": B})
+    custom_token.functions.approve(service_registry.address, deposit).call_and_transact(
+        {"from": B}
+    )
+    service_registry.functions.deposit(deposit).call_and_transact({"from": B})
+    assert service_registry.functions.hasValidRegistration(B).call()
+
```
Comment (pirapira) [inline raiden_contracts/tests/test_print_gas.py]: This blob is copied and pasted four times.  I should turn this into a function.

File: raiden_contracts/tests/test_one_to_n.py [hunk 2]
```diff
@@ -119,17 +121,57 @@ def test_claim(
...
+    deposit_to_udc: Callable,
+    get_accounts: Callable,
+    get_private_key: Callable,
+    web3: Web3,
+) -> None:
+    """OneToN contract should not work for an unregistered service provider."""
+    (A, B) = get_accounts(2)
+    deposit_to_udc(A, 30)
+
+    # happy case
```
Comment (palango) [inline raiden_contracts/tests/test_one_to_n.py]: why is this the happy case?

File: raiden_contracts/tests/test_one_to_n.py [hunk 3]
```diff
@@ -119,17 +121,57 @@ def test_claim(
...
+    amount = 10
+    expiration = web3.eth.blockNumber + 2
+    chain_id = int(web3.version.network)
```
Comment (palango) [inline raiden_contracts/tests/test_one_to_n.py]: this should be `web3.eth.chainId` after https://github.com/raiden-network/raiden-contracts/pull/1153

File: raiden_contracts/tests/test_print_gas.py [hunk 4]
```diff
@@ -260,6 +262,15 @@ def print_gas_monitoring_service(
     )
     service_registry.functions.deposit(SERVICE_DEPOSIT).call_and_transact({"from": MS})
 
+    # B registers itself as a service provider
+    deposit = service_registry.functions.currentPrice().call()
```
Comment (palango) [inline raiden_contracts/tests/test_print_gas.py]: Maybe you can refactor this out and use it in `create_service_account` as well to reduce duplication.

File: raiden_contracts/tests/test_one_to_n.py (hunk 2 above)
Comment (pirapira) [reply raiden_contracts/tests/test_one_to_n.py]: No, I copied and pasted the happiness but left the unhappy case only.

File: raiden_contracts/tests/test_print_gas.py (hunk 4 above)
Comment (pirapira) [reply raiden_contracts/tests/test_print_gas.py]: Turns out these lines can simply be removed.  It's all done in `create_service_account` above.

File: raiden_contracts/tests/test_print_gas.py [hunk 5]
```diff
@@ -316,11 +318,22 @@ def print_gas_one_to_n(
...
     print_gas: Callable,
+    service_registry: Contract,
+    custom_token: Contract,
 ) -> None:
     """ Abusing pytest to print gas cost of OneToN functions """
     (A, B) = get_accounts(2)
     deposit_to_udc(A, 30)
 
+    # B registers itself as a service provider
+    deposit = service_registry.functions.currentPrice().call()
```
Comment (palango) [inline raiden_contracts/tests/test_print_gas.py]: This still seems duplicated:
- https://github.com/raiden-network/raiden-contracts/pull/1151/files#diff-a0ec52e4288fe8c7b94e7d1fc9704dc6R397
- https://github.com/raiden-network/raiden-contracts/pull/1151/files#diff-7ecca76ac1d6a1de743191515e65a0e0R62

Not sure what your general approach is in this repo, as it is only 3 lines...

Comment (pirapira) [reply raiden_contracts/tests/test_print_gas.py]: The general approach is to reuse the fixture.  A fix is coming in this PR.

Comment (pirapira) [reply raiden_contracts/tests/test_print_gas.py]: I've removed this part.

There is something similar in the same file, but there the duplication is harder to remove.  I need the gas consumption of the deposit transaction.

**Pull request merged**